components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java

```diff
@@ -177,11 +178,38 @@ public WebappMetadata getStoppedWebapp(String webappFileName, String hostName) {
         return webappMetadata;
     }
 
+    private void setServiceListPath(WebApplication webApplication) {
+        String serviceListPathParamName = "service-list-path";
+        String serviceListPathParam =
+                webApplication.getContext().getServletContext().getInitParameter(serviceListPathParamName);
+        if ("".equals(serviceListPathParam) || serviceListPathParam == null) {
+            Map<String, ? extends ServletRegistration> servletRegs =
+                    webApplication.getContext().getServletContext().getServletRegistrations();
+            for (ServletRegistration servletReg : servletRegs.values()) {
+                serviceListPathParam = servletReg.getInitParameter(serviceListPathParamName);
+                if (!"".equals(serviceListPathParam) || serviceListPathParam != null) {
+                    break;
+                }
+            }
+        }
+        if ("".equals(serviceListPathParam) || serviceListPathParam == null) {
```

Comment:
Shouldn't be the null check be the first condition checked and then the empty string?